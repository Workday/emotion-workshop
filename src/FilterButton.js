/**
 * Steps:
 * 1. Within the props function, change the background color and text color of the buttons
 */

import React from 'react'
import styled from '@emotion/styled'

export const ButtonTypes = {
	primary: 'primary',
	secondary: 'secondary',
	default: 'default',
  }

const ButtonContainer = styled.a`
	margin: 3px;
	padding: 6px 10px;
	text-decoration: none;
	border: 1px solid transparent;
	border-radius: 3px;
	margin: 0 10px;
	font-weight: 600;
	${(props => {
		/* based on the button type, change the backgroundColor and text color  */
	}
	)}
`
function FilterButton({buttonType, onClick, href, text}) {
    return ( <ButtonContainer href={href} onClick={onClick} buttonType={buttonType}>{text}</ButtonContainer>)
  }

export default FilterButton
