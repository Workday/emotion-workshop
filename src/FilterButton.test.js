import React from 'react'
import renderer from 'react-test-renderer'
import serializer from 'jest-emotion'
import FilterButton from './FilterButton'

expect.addSnapshotSerializer(serializer)

test('renders with correct styles', () => {
  const tree = renderer.create(<FilterButton onClick={jest.fn()} />).toJSON()

  expect(tree).toMatchSnapshot()
})
