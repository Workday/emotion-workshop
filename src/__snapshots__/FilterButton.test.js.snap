// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with correct styles 1`] = `
.emotion-0 {
  margin: 3px;
  padding: 6px 10px;
  -webkit-text-decoration: none;
  text-decoration: none;
  border: 1px solid transparent;
  border-radius: 3px;
  margin: 0 10px;
  font-weight: 600;
  background-color: #e0e0e0;
  color: black;
}

<a
  className="emotion-0"
  onClick={[MockFunction]}
/>
`;
